extensions/arc/src/ui/tree/controllerTreeDataProvider.ts

```diff
@@ -0,0 +1,59 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the Source EULA. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import * as vscode from 'vscode';
+import { ControllerTreeNode } from './controllerTreeNode';
+import { TreeNode } from './treeNode';
+import { LoadingControllerNode as LoadingTreeNode } from './loadingTreeNode';
+import { ControllerModel } from '../../models/controllerModel';
+
+/**
+ * The TreeDataProvider for the Azure Arc view, which displays a list of registered
+ * controllers and the resources under them.
+ */
+export class AzureArcTreeDataProvider implements vscode.TreeDataProvider<TreeNode> {
+
+	private _onDidChangeTreeData: vscode.EventEmitter<TreeNode | undefined> = new vscode.EventEmitter<TreeNode | undefined>();
+	readonly onDidChangeTreeData: vscode.Event<TreeNode | undefined> = this._onDidChangeTreeData.event;
+
+	private _loading: boolean = true;
+	private _loadingNode = new LoadingTreeNode();
+
+	private _controllerNodes: ControllerTreeNode[] = [];
+
+	constructor(private _context: vscode.ExtensionContext) {
+		// TODO: 
```

Comment:
What is to be done here?  I assume it's related to the 5 second delay.

Reply:
Load the persisted controllers (one-time action done upon creation)

Reply:
(I didn't bother to write context here since I'm fixing that right now myself)